{
  "nbformat": 4,
  "nbformat_minor": 0,
  "metadata": {
    "colab": {
      "name": "open_targets.ipynb",
      "provenance": [],
      "collapsed_sections": []
    },
    "kernelspec": {
      "name": "python3",
      "display_name": "Python 3"
    }
  },
  "cells": [
    {
      "cell_type": "code",
      "metadata": {
        "id": "DQoBPVYywCKw"
      },
      "source": [
        "import requests\n",
        "import pandas as pd\n",
        "import json"
      ],
      "execution_count": 1,
      "outputs": []
    },
    {
      "cell_type": "code",
      "metadata": {
        "id": "3WzOWaRnwIAk"
      },
      "source": [
        "# query open targets and return json\n",
        "def gene_coloc_open_targets_query(geneid):\n",
        "  \"\"\"\n",
        "  queries open targets for colocalization data given gene id (ENS###)\n",
        "\n",
        "  input: geneid (ENS###)\n",
        "  output: coloc dataframe with the following columns:\n",
        "  phenotypeId\tqtlStudyId\th3\th4\tlog2h4h3\tstudy.studyId\tstudy.traitReported\t\n",
        "    study.pubAuthor\tleftVariant.id\ttissue.id\ttissue.name\n",
        "  \"\"\"\n",
        "  # query open targets for each gene in list\n",
        "  api_query = f'''query {{  \n",
        "    colocalisationsForGene(geneId: \"{geneid}\") {{\n",
        "      study{{\n",
        "        studyId\n",
        "        traitReported\n",
        "        pubAuthor\n",
        "      }}\n",
        "      leftVariant {{\n",
        "        id\n",
        "      }}\n",
        "      phenotypeId\n",
        "      tissue {{\n",
        "        id\n",
        "        name\n",
        "      }}\n",
        "      qtlStudyId\n",
        "      h3\n",
        "      h4\n",
        "      log2h4h3 \n",
        "    }}\n",
        "  }}'''\n",
        "\n",
        "  #set base_url for Open Targets Genetics Portal API\n",
        "  base_url = \"http://genetics-api.opentargets.io/graphql\"\n",
        "\n",
        "  #perform API call and check status code of response\n",
        "  r = requests.post(base_url, json={'query': api_query})\n",
        "  if str(r.status_code) == '400':\n",
        "    print(f'{geneid} query status code: {r.status_code}')\n",
        "  \n",
        "  else:\n",
        "    pass\n",
        "\n",
        "  #transform API response into JSON \n",
        "  api_response_as_json = json.loads(r.text)\n",
        "  \n",
        "  return api_response_as_json\n",
        "\n",
        "\n",
        "def qtl_coloc_open_targets_query(variant_id):\n",
        "  \"\"\"\n",
        "  queries open targets for qtl data given variant_id in chr_pos_a1_a2 format \n",
        "    (hg38)\n",
        "\n",
        "  input: variant id in chr_pos_a1_a2 format (hg38)\n",
        "  output: dataframe with the following columns:\n",
        "  qtls gene.symbol gene.id\n",
        "  \"\"\"\n",
        "  api_query = f'''query {{\n",
        "    genesForVariant(variantId: \"{variant_id}\") {{\n",
        "      gene {{\n",
        "        id\n",
        "        symbol\n",
        "      }}\n",
        "      qtls{{\n",
        "        typeId\n",
        "        sourceId\n",
        "        aggregatedScore\n",
        "        tissues{{\n",
        "          tissue {{\n",
        "            id\n",
        "            name\n",
        "          }}\n",
        "          quantile\n",
        "          beta\n",
        "          pval\n",
        "        }}\n",
        "      }}\n",
        "    }}\n",
        "  }}'''\n",
        "\n",
        "  #set base_url for Open Targets Genetics Portal API\n",
        "  base_url = \"http://genetics-api.opentargets.io/graphql\"\n",
        "\n",
        "  #perform API call and check status code of response\n",
        "  r = requests.post(base_url, json={'query': api_query})\n",
        "  if str(r.status_code) == '400':\n",
        "    print(f'{variant_id} query status code: {r.status_code}')\n",
        "  \n",
        "  else:\n",
        "    pass\n",
        "\n",
        "  #transform API response into JSON \n",
        "  api_response_as_json = json.loads(r.text)\n",
        "  \n",
        "  return api_response_as_json\n",
        "\n",
        "\n",
        "def phewas_open_targets_query(variant_id):\n",
        "  \"\"\"\n",
        "  query open targets for phewas data\n",
        "\n",
        "  input: variant id (chr_pos_a1_a2 format with hg38 positions)\n",
        "  output: phewas dataframe with the following columns:\n",
        "  pval\tbeta\tnTotal\tnCases\toddsRatio\teaf\tse\tstudy.studyId\t\n",
        "    study.traitReported\tstudy.pubAuthor\n",
        "  \"\"\"\n",
        "  api_query = \"\"\"\n",
        "    query pheWAS($variantId: String!){\n",
        "      pheWAS(variantId: $variantId) {\n",
        "        associations {\n",
        "          study {\n",
        "            studyId\n",
        "            traitReported\n",
        "            pubAuthor\n",
        "          }\n",
        "          pval\n",
        "          beta\n",
        "          nTotal\n",
        "          nCases\n",
        "          oddsRatio\n",
        "          eaf\n",
        "          se\n",
        "        }\n",
        "      }\n",
        "    }\"\"\" \n",
        "\n",
        "  #set base_url for Open Targets Genetics Portal API\n",
        "  base_url = \"http://genetics-api.opentargets.io/graphql\"\n",
        "  #set variables object\n",
        "  variables = {\"variantId\": variant_id}\n",
        "\n",
        "  #perform API call and check status code of response\n",
        "  r = requests.post(base_url, json={'query':api_query, \"variables\":variables})\n",
        "  if str(r.status_code) == '400':\n",
        "    print(f'{variant_id} query status code: {r.status_code}')\n",
        "  \n",
        "  else:\n",
        "    pass\n",
        "\n",
        "  #transform API response into JSON \n",
        "  api_response_as_json = json.loads(r.text)\n",
        "  \n",
        "  return api_response_as_json\n",
        "\n",
        "\n",
        "def search_rsid_open_targets_query(rsid):\n",
        "\n",
        "  \"\"\"\n",
        "  query open targets for snp ids in chr_pos_a1_a2 and allele freqs with rsid\n",
        "\n",
        "  input: rsid\n",
        "  output: ids and allele frequencies for different ancestry groups with columns:\n",
        "  id\trsId\tgnomadAFR\tgnomadAMR\tgnomadASJ\tgnomadEAS\tgnomadFIN\tgnomadNFE\t\n",
        "    gnomadNFEEST\tgnomadNFENWE\tgnomadNFESEU\tgnomadNFEONF\tgnomadOTH\n",
        "\n",
        "  Some information on the allele frequencies being pulled \n",
        "\n",
        "  \"gnomAD Allele frequency (African/African-American population)\"\n",
        "    gnomadAFR: Float\n",
        "\n",
        "    \"gnomAD Allele frequency (Latino/Admixed American population)\"\n",
        "    gnomadAMR: Float\n",
        "\n",
        "    \"gnomAD Allele frequency (Ashkenazi Jewish population)\"\n",
        "    gnomadASJ: Float\n",
        "\n",
        "    \"gnomAD Allele frequency (East Asian population)\"\n",
        "    gnomadEAS: Float\n",
        "\n",
        "    \"gnomAD Allele frequency (Finnish population)\"\n",
        "    gnomadFIN: Float\n",
        "\n",
        "    \"gnomAD Allele frequency (Non-Finnish European population)\"\n",
        "    gnomadNFE: Float\n",
        "\n",
        "    \"gnomAD Allele frequency (Non-Finnish Eurpoean Estonian sub-population)\"\n",
        "    gnomadNFEEST: Float\n",
        "\n",
        "    \"gnomAD Allele frequency (Non-Finnish Eurpoean North-Western European sub-population)\"\n",
        "    gnomadNFENWE: Float\n",
        "\n",
        "    \"gnomAD Allele frequency (Non-Finnish Eurpoean Southern European sub-population)\"\n",
        "    gnomadNFESEU: Float\n",
        "\n",
        "    \"gnomAD Allele frequency (Non-Finnish Eurpoean Other non-Finnish European sub-population)\"\n",
        "    gnomadNFEONF: Float\n",
        "\n",
        "    \"gnomAD Allele frequency (Other (population not assigned) population)\"\n",
        "    gnomadOTH: Float\n",
        "    \"\"\"\n",
        "\n",
        "  api_query = \"\"\"\n",
        "  query search($queryString: String!){\n",
        "    search(queryString: $queryString) {\n",
        "      variants{\n",
        "        id\n",
        "        rsId\n",
        "        gnomadAFR\n",
        "        gnomadAMR\n",
        "        gnomadASJ\n",
        "        gnomadEAS\n",
        "        gnomadFIN\n",
        "        gnomadNFE\n",
        "        gnomadNFEEST\n",
        "        gnomadNFENWE\n",
        "        gnomadNFESEU\n",
        "        gnomadNFEONF\n",
        "        gnomadOTH\n",
        "      }\n",
        "    }\n",
        "  }\"\"\"\n",
        "\n",
        "  #set base_url for Open Targets Genetics Portal API\n",
        "  base_url = \"http://genetics-api.opentargets.io/graphql\"\n",
        "  #set variables object\n",
        "  variables = {\"queryString\": rsid}\n",
        "\n",
        "  #perform API call and check status code of response\n",
        "  r = requests.post(base_url, json={'query':api_query, \"variables\":variables})\n",
        "  if str(r.status_code) == '400':\n",
        "    print(f'{rsid} query status code: {r.status_code}')\n",
        "  \n",
        "  else:\n",
        "    pass\n",
        "\n",
        "  #transform API response into JSON \n",
        "  api_response_as_json = json.loads(r.text)\n",
        "  \n",
        "  return api_response_as_json\n",
        "\n",
        "\n",
        "def query_qtls(snp_list):\n",
        "  \"\"\"\n",
        "  uses qtl_coloc_open_targets_query() function to query open targets for qtls\n",
        "    given a list of snps in chr_pos_a1_a2 format (hg38)\n",
        "\n",
        "  input: list of snps (chr_pos_a1_a2 formatted with hg38 positions)\n",
        "  output: cleaned up dataframe of qtls with following columns:\n",
        "  snp\tgene_symbol\tgene_id\ttype\ttissue\tbeta\tpval\n",
        "  \"\"\"\n",
        "\n",
        "  total_qtls_df = pd.DataFrame()\n",
        "  for snp in snp_list:\n",
        "    qtl_query = qtl_coloc_open_targets_query(snp)\n",
        "    qtl_query_df = pd.json_normalize(qtl_query['data']['genesForVariant'])\n",
        "    if len(qtl_query_df) !=  0:\n",
        "      qtl_query_df['snp'] = snp\n",
        "      qtls_df = qtl_query_df.loc[qtl_query_df.qtls.map(lambda d: len(d)) > 0].reset_index(drop=True).copy()\n",
        "      total_qtls_df = total_qtls_df.append(qtls_df)\n",
        "  total_qtls_df = total_qtls_df.reset_index(drop=True)\n",
        "\n",
        "  final_qtls_df = pd.DataFrame()\n",
        "  for i, qtl in enumerate(total_qtls_df.qtls):\n",
        "    for j, qt in enumerate(qtl):\n",
        "      for k, tissue in enumerate(qt['tissues']):\n",
        "\n",
        "        qtl_dict = {\n",
        "            'snp': total_qtls_df.loc[i,'snp'],\n",
        "            'gene_symbol': total_qtls_df.loc[i,'gene.symbol'],\n",
        "            'gene_id': total_qtls_df.loc[i,'gene.id'],\n",
        "            'type': qt['typeId'],\n",
        "            'tissue': tissue['tissue']['name'],\n",
        "            'beta': tissue['beta'],\n",
        "            'pval': tissue['pval']\n",
        "        }\n",
        "        \n",
        "        qtl_dict_df = pd.DataFrame.from_records([qtl_dict])\n",
        "        final_qtls_df = final_qtls_df.append(qtl_dict_df).reset_index(drop=True)\n",
        "        \n",
        "  return final_qtls_df"
      ],
      "execution_count": 2,
      "outputs": []
    },
    {
      "cell_type": "code",
      "metadata": {
        "id": "SFm7D3mZbEAq",
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 204
        },
        "outputId": "0aba9fc7-7a40-456d-9187-6b21c6e1a84f"
      },
      "source": [
        "# query coloc data given gene_id\n",
        "coloc_query = gene_coloc_open_targets_query('ENSG00000160714')\n",
        "coloc_query_df = pd.json_normalize(coloc_query['data']['colocalisationsForGene'])\n",
        "coloc_query_df.head()"
      ],
      "execution_count": 3,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/html": [
              "<div>\n",
              "<style scoped>\n",
              "    .dataframe tbody tr th:only-of-type {\n",
              "        vertical-align: middle;\n",
              "    }\n",
              "\n",
              "    .dataframe tbody tr th {\n",
              "        vertical-align: top;\n",
              "    }\n",
              "\n",
              "    .dataframe thead th {\n",
              "        text-align: right;\n",
              "    }\n",
              "</style>\n",
              "<table border=\"1\" class=\"dataframe\">\n",
              "  <thead>\n",
              "    <tr style=\"text-align: right;\">\n",
              "      <th></th>\n",
              "      <th>phenotypeId</th>\n",
              "      <th>qtlStudyId</th>\n",
              "      <th>h3</th>\n",
              "      <th>h4</th>\n",
              "      <th>log2h4h3</th>\n",
              "      <th>study.studyId</th>\n",
              "      <th>study.traitReported</th>\n",
              "      <th>study.pubAuthor</th>\n",
              "      <th>leftVariant.id</th>\n",
              "      <th>tissue.id</th>\n",
              "      <th>tissue.name</th>\n",
              "    </tr>\n",
              "  </thead>\n",
              "  <tbody>\n",
              "    <tr>\n",
              "      <th>0</th>\n",
              "      <td>ENSG00000160714</td>\n",
              "      <td>QUACH_2016</td>\n",
              "      <td>0.183847</td>\n",
              "      <td>0.238134</td>\n",
              "      <td>0.373264</td>\n",
              "      <td>NEALE2_2395_1</td>\n",
              "      <td>Pattern 1 | hair/balding pattern</td>\n",
              "      <td>UKB Neale v2</td>\n",
              "      <td>1_153901213_C_CA</td>\n",
              "      <td>MONOCYTE_IAV</td>\n",
              "      <td>Monocyte iav</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>1</th>\n",
              "      <td>ENSG00000160714</td>\n",
              "      <td>QUACH_2016</td>\n",
              "      <td>0.222941</td>\n",
              "      <td>0.116175</td>\n",
              "      <td>-0.940364</td>\n",
              "      <td>NEALE2_1050</td>\n",
              "      <td>Time spend outdoors in summer</td>\n",
              "      <td>UKB Neale v2</td>\n",
              "      <td>1_153987528_A_AAAAG</td>\n",
              "      <td>MONOCYTE_IAV</td>\n",
              "      <td>Monocyte iav</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>2</th>\n",
              "      <td>ENSG00000160714</td>\n",
              "      <td>QUACH_2016</td>\n",
              "      <td>0.152196</td>\n",
              "      <td>0.422189</td>\n",
              "      <td>1.471959</td>\n",
              "      <td>NEALE2_30040_raw</td>\n",
              "      <td>Mean corpuscular volume</td>\n",
              "      <td>UKB Neale v2</td>\n",
              "      <td>1_154078049_A_G</td>\n",
              "      <td>MONOCYTE_IAV</td>\n",
              "      <td>Monocyte iav</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>3</th>\n",
              "      <td>ENSG00000160714</td>\n",
              "      <td>QUACH_2016</td>\n",
              "      <td>0.199173</td>\n",
              "      <td>0.239907</td>\n",
              "      <td>0.268452</td>\n",
              "      <td>NEALE2_30050_raw</td>\n",
              "      <td>Mean corpuscular haemoglobin</td>\n",
              "      <td>UKB Neale v2</td>\n",
              "      <td>1_154095720_C_T</td>\n",
              "      <td>MONOCYTE_IAV</td>\n",
              "      <td>Monocyte iav</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>4</th>\n",
              "      <td>ENSG00000160714</td>\n",
              "      <td>QUACH_2016</td>\n",
              "      <td>0.208134</td>\n",
              "      <td>0.210694</td>\n",
              "      <td>0.017637</td>\n",
              "      <td>NEALE2_864</td>\n",
              "      <td>Number of days/week walked 10+ minutes</td>\n",
              "      <td>UKB Neale v2</td>\n",
              "      <td>1_154113167_A_T</td>\n",
              "      <td>MONOCYTE_IAV</td>\n",
              "      <td>Monocyte iav</td>\n",
              "    </tr>\n",
              "  </tbody>\n",
              "</table>\n",
              "</div>"
            ],
            "text/plain": [
              "       phenotypeId  qtlStudyId  ...     tissue.id   tissue.name\n",
              "0  ENSG00000160714  QUACH_2016  ...  MONOCYTE_IAV  Monocyte iav\n",
              "1  ENSG00000160714  QUACH_2016  ...  MONOCYTE_IAV  Monocyte iav\n",
              "2  ENSG00000160714  QUACH_2016  ...  MONOCYTE_IAV  Monocyte iav\n",
              "3  ENSG00000160714  QUACH_2016  ...  MONOCYTE_IAV  Monocyte iav\n",
              "4  ENSG00000160714  QUACH_2016  ...  MONOCYTE_IAV  Monocyte iav\n",
              "\n",
              "[5 rows x 11 columns]"
            ]
          },
          "metadata": {
            "tags": []
          },
          "execution_count": 3
        }
      ]
    },
    {
      "cell_type": "code",
      "metadata": {
        "id": "PVO_iKlwwSdO",
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 855
        },
        "outputId": "6f31423d-87c3-487e-ca9a-9be1b82b2d5f"
      },
      "source": [
        "# query qtls for a list of snps\n",
        "# query_qtls() uses qtl_coloc_open_targets_query() for each snp id in input list and returns combined and cleaned up dataframe\n",
        "test_snp_list = ['1_154453788_C_T', '1_154445939_T_C']\n",
        "qtls_df = query_qtls(test_snp_list)\n",
        "qtls_df.head()"
      ],
      "execution_count": 4,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/html": [
              "<div>\n",
              "<style scoped>\n",
              "    .dataframe tbody tr th:only-of-type {\n",
              "        vertical-align: middle;\n",
              "    }\n",
              "\n",
              "    .dataframe tbody tr th {\n",
              "        vertical-align: top;\n",
              "    }\n",
              "\n",
              "    .dataframe thead th {\n",
              "        text-align: right;\n",
              "    }\n",
              "</style>\n",
              "<table border=\"1\" class=\"dataframe\">\n",
              "  <thead>\n",
              "    <tr style=\"text-align: right;\">\n",
              "      <th></th>\n",
              "      <th>snp</th>\n",
              "      <th>gene_symbol</th>\n",
              "      <th>gene_id</th>\n",
              "      <th>type</th>\n",
              "      <th>tissue</th>\n",
              "      <th>beta</th>\n",
              "      <th>pval</th>\n",
              "    </tr>\n",
              "  </thead>\n",
              "  <tbody>\n",
              "    <tr>\n",
              "      <th>0</th>\n",
              "      <td>1_154453788_C_T</td>\n",
              "      <td>UBE2Q1</td>\n",
              "      <td>ENSG00000160714</td>\n",
              "      <td>eqtl</td>\n",
              "      <td>Blood (eQTLGen)</td>\n",
              "      <td>-0.047601</td>\n",
              "      <td>8.819400e-09</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>1</th>\n",
              "      <td>1_154453788_C_T</td>\n",
              "      <td>IL6R</td>\n",
              "      <td>ENSG00000160712</td>\n",
              "      <td>eqtl</td>\n",
              "      <td>Blood (GTEX v7)</td>\n",
              "      <td>-0.106938</td>\n",
              "      <td>3.641230e-07</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>2</th>\n",
              "      <td>1_154453788_C_T</td>\n",
              "      <td>IL6R</td>\n",
              "      <td>ENSG00000160712</td>\n",
              "      <td>eqtl</td>\n",
              "      <td>Macrophage listeria (NEDELEC 2016)</td>\n",
              "      <td>0.153303</td>\n",
              "      <td>1.443160e-06</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>3</th>\n",
              "      <td>1_154453788_C_T</td>\n",
              "      <td>IL6R</td>\n",
              "      <td>ENSG00000160712</td>\n",
              "      <td>eqtl</td>\n",
              "      <td>Monocyte r848 (QUACH 2016)</td>\n",
              "      <td>0.119603</td>\n",
              "      <td>2.832830e-07</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>4</th>\n",
              "      <td>1_154453788_C_T</td>\n",
              "      <td>IL6R</td>\n",
              "      <td>ENSG00000160712</td>\n",
              "      <td>eqtl</td>\n",
              "      <td>Colon transverse (GTEX v7)</td>\n",
              "      <td>-0.269436</td>\n",
              "      <td>7.835600e-06</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>5</th>\n",
              "      <td>1_154453788_C_T</td>\n",
              "      <td>IL6R</td>\n",
              "      <td>ENSG00000160712</td>\n",
              "      <td>eqtl</td>\n",
              "      <td>Monocyte pam3csk4 (QUACH 2016)</td>\n",
              "      <td>0.119035</td>\n",
              "      <td>1.144180e-07</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>6</th>\n",
              "      <td>1_154453788_C_T</td>\n",
              "      <td>IL6R</td>\n",
              "      <td>ENSG00000160712</td>\n",
              "      <td>eqtl</td>\n",
              "      <td>Monocyte lps (QUACH 2016)</td>\n",
              "      <td>0.116618</td>\n",
              "      <td>1.808160e-07</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>7</th>\n",
              "      <td>1_154453788_C_T</td>\n",
              "      <td>IL6R</td>\n",
              "      <td>ENSG00000160712</td>\n",
              "      <td>eqtl</td>\n",
              "      <td>Artery tibial (GTEX v7)</td>\n",
              "      <td>-0.178372</td>\n",
              "      <td>2.417210e-09</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>8</th>\n",
              "      <td>1_154453788_C_T</td>\n",
              "      <td>IL6R</td>\n",
              "      <td>ENSG00000160712</td>\n",
              "      <td>eqtl</td>\n",
              "      <td>Esophagus muscularis (GTEX v7)</td>\n",
              "      <td>-0.231648</td>\n",
              "      <td>1.751990e-06</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>9</th>\n",
              "      <td>1_154453788_C_T</td>\n",
              "      <td>IL6R</td>\n",
              "      <td>ENSG00000160712</td>\n",
              "      <td>pqtl</td>\n",
              "      <td>Blood plasma (SUN2018)</td>\n",
              "      <td>1.114800</td>\n",
              "      <td>0.000000e+00</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>10</th>\n",
              "      <td>1_154453788_C_T</td>\n",
              "      <td>IL6R</td>\n",
              "      <td>ENSG00000160712</td>\n",
              "      <td>pqtl</td>\n",
              "      <td>Blood plasma (SUN2018)</td>\n",
              "      <td>1.102300</td>\n",
              "      <td>0.000000e+00</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>11</th>\n",
              "      <td>1_154453788_C_T</td>\n",
              "      <td>TDRD10</td>\n",
              "      <td>ENSG00000163239</td>\n",
              "      <td>eqtl</td>\n",
              "      <td>Adipose visceral omentum (GTEX v7)</td>\n",
              "      <td>0.194820</td>\n",
              "      <td>2.137440e-07</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>12</th>\n",
              "      <td>1_154453788_C_T</td>\n",
              "      <td>TDRD10</td>\n",
              "      <td>ENSG00000163239</td>\n",
              "      <td>eqtl</td>\n",
              "      <td>Esophagus muscularis (GTEX v7)</td>\n",
              "      <td>0.176498</td>\n",
              "      <td>5.156840e-07</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>13</th>\n",
              "      <td>1_154453788_C_T</td>\n",
              "      <td>NUP210L</td>\n",
              "      <td>ENSG00000143552</td>\n",
              "      <td>eqtl</td>\n",
              "      <td>Blood (eQTLGen)</td>\n",
              "      <td>-0.040033</td>\n",
              "      <td>1.435600e-06</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>14</th>\n",
              "      <td>1_154445939_T_C</td>\n",
              "      <td>UBE2Q1</td>\n",
              "      <td>ENSG00000160714</td>\n",
              "      <td>eqtl</td>\n",
              "      <td>Blood (eQTLGen)</td>\n",
              "      <td>-0.046115</td>\n",
              "      <td>1.663400e-08</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>15</th>\n",
              "      <td>1_154445939_T_C</td>\n",
              "      <td>IL6R</td>\n",
              "      <td>ENSG00000160712</td>\n",
              "      <td>eqtl</td>\n",
              "      <td>Macrophage listeria (NEDELEC 2016)</td>\n",
              "      <td>0.177325</td>\n",
              "      <td>1.420520e-07</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>16</th>\n",
              "      <td>1_154445939_T_C</td>\n",
              "      <td>IL6R</td>\n",
              "      <td>ENSG00000160712</td>\n",
              "      <td>eqtl</td>\n",
              "      <td>Monocyte r848 (QUACH 2016)</td>\n",
              "      <td>0.115671</td>\n",
              "      <td>7.995590e-07</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>17</th>\n",
              "      <td>1_154445939_T_C</td>\n",
              "      <td>IL6R</td>\n",
              "      <td>ENSG00000160712</td>\n",
              "      <td>eqtl</td>\n",
              "      <td>Colon transverse (GTEX v7)</td>\n",
              "      <td>-0.322242</td>\n",
              "      <td>4.274030e-08</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>18</th>\n",
              "      <td>1_154445939_T_C</td>\n",
              "      <td>IL6R</td>\n",
              "      <td>ENSG00000160712</td>\n",
              "      <td>eqtl</td>\n",
              "      <td>Monocyte pam3csk4 (QUACH 2016)</td>\n",
              "      <td>0.115649</td>\n",
              "      <td>3.021910e-07</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>19</th>\n",
              "      <td>1_154445939_T_C</td>\n",
              "      <td>IL6R</td>\n",
              "      <td>ENSG00000160712</td>\n",
              "      <td>eqtl</td>\n",
              "      <td>Monocyte lps (QUACH 2016)</td>\n",
              "      <td>0.116038</td>\n",
              "      <td>2.281460e-07</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>20</th>\n",
              "      <td>1_154445939_T_C</td>\n",
              "      <td>IL6R</td>\n",
              "      <td>ENSG00000160712</td>\n",
              "      <td>eqtl</td>\n",
              "      <td>Artery tibial (GTEX v7)</td>\n",
              "      <td>-0.181389</td>\n",
              "      <td>4.920270e-09</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>21</th>\n",
              "      <td>1_154445939_T_C</td>\n",
              "      <td>IL6R</td>\n",
              "      <td>ENSG00000160712</td>\n",
              "      <td>eqtl</td>\n",
              "      <td>Blood (eQTLGen)</td>\n",
              "      <td>-0.038970</td>\n",
              "      <td>1.709900e-06</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>22</th>\n",
              "      <td>1_154445939_T_C</td>\n",
              "      <td>IL6R</td>\n",
              "      <td>ENSG00000160712</td>\n",
              "      <td>eqtl</td>\n",
              "      <td>Esophagus muscularis (GTEX v7)</td>\n",
              "      <td>-0.235512</td>\n",
              "      <td>2.412090e-06</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>23</th>\n",
              "      <td>1_154445939_T_C</td>\n",
              "      <td>IL6R</td>\n",
              "      <td>ENSG00000160712</td>\n",
              "      <td>pqtl</td>\n",
              "      <td>Blood plasma (SUN2018)</td>\n",
              "      <td>1.075600</td>\n",
              "      <td>0.000000e+00</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>24</th>\n",
              "      <td>1_154445939_T_C</td>\n",
              "      <td>IL6R</td>\n",
              "      <td>ENSG00000160712</td>\n",
              "      <td>pqtl</td>\n",
              "      <td>Blood plasma (SUN2018)</td>\n",
              "      <td>1.063200</td>\n",
              "      <td>0.000000e+00</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>25</th>\n",
              "      <td>1_154445939_T_C</td>\n",
              "      <td>NUP210L</td>\n",
              "      <td>ENSG00000143552</td>\n",
              "      <td>eqtl</td>\n",
              "      <td>Blood (eQTLGen)</td>\n",
              "      <td>-0.040704</td>\n",
              "      <td>6.585700e-07</td>\n",
              "    </tr>\n",
              "  </tbody>\n",
              "</table>\n",
              "</div>"
            ],
            "text/plain": [
              "                snp gene_symbol  ...      beta          pval\n",
              "0   1_154453788_C_T      UBE2Q1  ... -0.047601  8.819400e-09\n",
              "1   1_154453788_C_T        IL6R  ... -0.106938  3.641230e-07\n",
              "2   1_154453788_C_T        IL6R  ...  0.153303  1.443160e-06\n",
              "3   1_154453788_C_T        IL6R  ...  0.119603  2.832830e-07\n",
              "4   1_154453788_C_T        IL6R  ... -0.269436  7.835600e-06\n",
              "5   1_154453788_C_T        IL6R  ...  0.119035  1.144180e-07\n",
              "6   1_154453788_C_T        IL6R  ...  0.116618  1.808160e-07\n",
              "7   1_154453788_C_T        IL6R  ... -0.178372  2.417210e-09\n",
              "8   1_154453788_C_T        IL6R  ... -0.231648  1.751990e-06\n",
              "9   1_154453788_C_T        IL6R  ...  1.114800  0.000000e+00\n",
              "10  1_154453788_C_T        IL6R  ...  1.102300  0.000000e+00\n",
              "11  1_154453788_C_T      TDRD10  ...  0.194820  2.137440e-07\n",
              "12  1_154453788_C_T      TDRD10  ...  0.176498  5.156840e-07\n",
              "13  1_154453788_C_T     NUP210L  ... -0.040033  1.435600e-06\n",
              "14  1_154445939_T_C      UBE2Q1  ... -0.046115  1.663400e-08\n",
              "15  1_154445939_T_C        IL6R  ...  0.177325  1.420520e-07\n",
              "16  1_154445939_T_C        IL6R  ...  0.115671  7.995590e-07\n",
              "17  1_154445939_T_C        IL6R  ... -0.322242  4.274030e-08\n",
              "18  1_154445939_T_C        IL6R  ...  0.115649  3.021910e-07\n",
              "19  1_154445939_T_C        IL6R  ...  0.116038  2.281460e-07\n",
              "20  1_154445939_T_C        IL6R  ... -0.181389  4.920270e-09\n",
              "21  1_154445939_T_C        IL6R  ... -0.038970  1.709900e-06\n",
              "22  1_154445939_T_C        IL6R  ... -0.235512  2.412090e-06\n",
              "23  1_154445939_T_C        IL6R  ...  1.075600  0.000000e+00\n",
              "24  1_154445939_T_C        IL6R  ...  1.063200  0.000000e+00\n",
              "25  1_154445939_T_C     NUP210L  ... -0.040704  6.585700e-07\n",
              "\n",
              "[26 rows x 7 columns]"
            ]
          },
          "metadata": {
            "tags": []
          },
          "execution_count": 4
        }
      ]
    },
    {
      "cell_type": "code",
      "metadata": {
        "id": "lrnLMv-7H7G7",
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 204
        },
        "outputId": "876aba7c-d307-4a1b-8cd7-c9dae38b9784"
      },
      "source": [
        "# query phewas data for a given snp\n",
        "phewas_query = phewas_open_targets_query('1_154453788_C_T')\n",
        "phewas_query_df = pd.json_normalize(phewas_query['data']['pheWAS']['associations'])\n",
        "phewas_query_df.head()"
      ],
      "execution_count": 5,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/html": [
              "<div>\n",
              "<style scoped>\n",
              "    .dataframe tbody tr th:only-of-type {\n",
              "        vertical-align: middle;\n",
              "    }\n",
              "\n",
              "    .dataframe tbody tr th {\n",
              "        vertical-align: top;\n",
              "    }\n",
              "\n",
              "    .dataframe thead th {\n",
              "        text-align: right;\n",
              "    }\n",
              "</style>\n",
              "<table border=\"1\" class=\"dataframe\">\n",
              "  <thead>\n",
              "    <tr style=\"text-align: right;\">\n",
              "      <th></th>\n",
              "      <th>pval</th>\n",
              "      <th>beta</th>\n",
              "      <th>nTotal</th>\n",
              "      <th>nCases</th>\n",
              "      <th>oddsRatio</th>\n",
              "      <th>eaf</th>\n",
              "      <th>se</th>\n",
              "      <th>study.studyId</th>\n",
              "      <th>study.traitReported</th>\n",
              "      <th>study.pubAuthor</th>\n",
              "    </tr>\n",
              "  </thead>\n",
              "  <tbody>\n",
              "    <tr>\n",
              "      <th>0</th>\n",
              "      <td>1.225010e-07</td>\n",
              "      <td>0.067727</td>\n",
              "      <td>344729</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>0.409706</td>\n",
              "      <td>0.012803</td>\n",
              "      <td>NEALE2_30270_raw</td>\n",
              "      <td>Mean sphered cell volume</td>\n",
              "      <td>UKB Neale v2</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>1</th>\n",
              "      <td>1.356400e-02</td>\n",
              "      <td>0.021674</td>\n",
              "      <td>36919</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>0.411766</td>\n",
              "      <td>0.008779</td>\n",
              "      <td>NEALE2_1001</td>\n",
              "      <td>Duration of strenuous sports</td>\n",
              "      <td>UKB Neale v2</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>2</th>\n",
              "      <td>1.837000e-04</td>\n",
              "      <td>-0.060700</td>\n",
              "      <td>40266</td>\n",
              "      <td>12194.0</td>\n",
              "      <td>0.941106</td>\n",
              "      <td>0.375928</td>\n",
              "      <td>0.016200</td>\n",
              "      <td>GCST004132</td>\n",
              "      <td>Crohn's disease</td>\n",
              "      <td>de Lange KM</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>3</th>\n",
              "      <td>3.510000e-02</td>\n",
              "      <td>0.106000</td>\n",
              "      <td>402099</td>\n",
              "      <td>820.0</td>\n",
              "      <td>1.111822</td>\n",
              "      <td>0.409935</td>\n",
              "      <td>0.050301</td>\n",
              "      <td>SAIGE_743_9</td>\n",
              "      <td>Osteopenia or other disorder of bone and carti...</td>\n",
              "      <td>UKB SAIGE</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>4</th>\n",
              "      <td>1.930000e-02</td>\n",
              "      <td>0.443000</td>\n",
              "      <td>373092</td>\n",
              "      <td>58.0</td>\n",
              "      <td>1.557372</td>\n",
              "      <td>0.410034</td>\n",
              "      <td>0.189003</td>\n",
              "      <td>SAIGE_272_12</td>\n",
              "      <td>Hyperglyceridemia</td>\n",
              "      <td>UKB SAIGE</td>\n",
              "    </tr>\n",
              "  </tbody>\n",
              "</table>\n",
              "</div>"
            ],
            "text/plain": [
              "           pval  ...  study.pubAuthor\n",
              "0  1.225010e-07  ...     UKB Neale v2\n",
              "1  1.356400e-02  ...     UKB Neale v2\n",
              "2  1.837000e-04  ...      de Lange KM\n",
              "3  3.510000e-02  ...        UKB SAIGE\n",
              "4  1.930000e-02  ...        UKB SAIGE\n",
              "\n",
              "[5 rows x 10 columns]"
            ]
          },
          "metadata": {
            "tags": []
          },
          "execution_count": 5
        }
      ]
    },
    {
      "cell_type": "code",
      "metadata": {
        "id": "C0dvAkM_CLV5",
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 80
        },
        "outputId": "2a4ba5af-9774-4904-a1af-b4056a267f49"
      },
      "source": [
        "# get hg38 chr_pos_a1_a2 and allele freqs for given rsid\n",
        "rsid_query = search_rsid_open_targets_query(\"rs4129267\")\n",
        "rsid_query_df = pd.json_normalize(rsid_query['data']['search']['variants'])\n",
        "rsid_query_df.head()"
      ],
      "execution_count": 6,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/html": [
              "<div>\n",
              "<style scoped>\n",
              "    .dataframe tbody tr th:only-of-type {\n",
              "        vertical-align: middle;\n",
              "    }\n",
              "\n",
              "    .dataframe tbody tr th {\n",
              "        vertical-align: top;\n",
              "    }\n",
              "\n",
              "    .dataframe thead th {\n",
              "        text-align: right;\n",
              "    }\n",
              "</style>\n",
              "<table border=\"1\" class=\"dataframe\">\n",
              "  <thead>\n",
              "    <tr style=\"text-align: right;\">\n",
              "      <th></th>\n",
              "      <th>id</th>\n",
              "      <th>rsId</th>\n",
              "      <th>gnomadAFR</th>\n",
              "      <th>gnomadAMR</th>\n",
              "      <th>gnomadASJ</th>\n",
              "      <th>gnomadEAS</th>\n",
              "      <th>gnomadFIN</th>\n",
              "      <th>gnomadNFE</th>\n",
              "      <th>gnomadNFEEST</th>\n",
              "      <th>gnomadNFENWE</th>\n",
              "      <th>gnomadNFESEU</th>\n",
              "      <th>gnomadNFEONF</th>\n",
              "      <th>gnomadOTH</th>\n",
              "    </tr>\n",
              "  </thead>\n",
              "  <tbody>\n",
              "    <tr>\n",
              "      <th>0</th>\n",
              "      <td>1_154453788_C_T</td>\n",
              "      <td>rs4129267</td>\n",
              "      <td>0.137483</td>\n",
              "      <td>0.509434</td>\n",
              "      <td>0.451724</td>\n",
              "      <td>0.379534</td>\n",
              "      <td>0.301957</td>\n",
              "      <td>0.375928</td>\n",
              "      <td>0.340299</td>\n",
              "      <td>0.389706</td>\n",
              "      <td>0.528302</td>\n",
              "      <td>0.388941</td>\n",
              "      <td>0.331801</td>\n",
              "    </tr>\n",
              "  </tbody>\n",
              "</table>\n",
              "</div>"
            ],
            "text/plain": [
              "                id       rsId  gnomadAFR  ...  gnomadNFESEU  gnomadNFEONF  gnomadOTH\n",
              "0  1_154453788_C_T  rs4129267   0.137483  ...      0.528302      0.388941   0.331801\n",
              "\n",
              "[1 rows x 13 columns]"
            ]
          },
          "metadata": {
            "tags": []
          },
          "execution_count": 6
        }
      ]
    },
    {
      "cell_type": "code",
      "metadata": {
        "id": "XGWUOfEBCR1E"
      },
      "source": [
        ""
      ],
      "execution_count": 6,
      "outputs": []
    },
    {
      "cell_type": "code",
      "metadata": {
        "id": "GSYA41ZSaPP0"
      },
      "source": [
        ""
      ],
      "execution_count": 6,
      "outputs": []
    },
    {
      "cell_type": "code",
      "metadata": {
        "id": "JJNk5He-RHbf"
      },
      "source": [
        ""
      ],
      "execution_count": 6,
      "outputs": []
    },
    {
      "cell_type": "code",
      "metadata": {
        "id": "5jlNfc9TQJBU"
      },
      "source": [
        ""
      ],
      "execution_count": 6,
      "outputs": []
    },
    {
      "cell_type": "code",
      "metadata": {
        "id": "Q1HwW8kORgI-"
      },
      "source": [
        ""
      ],
      "execution_count": 6,
      "outputs": []
    }
  ]
}